src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2133,11 +2131,12 @@ internal override bool KnownCircularStruct
                     else
                     {
                         var diagnostics = BindingDiagnosticBag.GetInstance();
+                        Debug.Assert(diagnostics.DiagnosticBag is not null);
                         var value = (int)CheckStructCircularity(diagnostics).ToThreeState();
 
                         if (Interlocked.CompareExchange(ref _lazyKnownCircularStruct, value, (int)ThreeState.Unknown) == (int)ThreeState.Unknown)
                         {
-                            AddDeclarationDiagnostics(diagnostics);
+                            DeclaringCompilation.CircularStructDiagnostics.AddRange(diagnostics.DiagnosticBag);
```

Comment:
>diagnostics.DiagnosticBag [](http://example.com/codeflow?start=84&length=25)

Are we certain we couldn't record any dependencies in ```diagnostics```?

Reply:
I'm not familiar with what `DependenciesBag` is, but it seems you're looking for this?

`DeclaringCompilation.AddUsedAssemblies(diagnostics.DependenciesBag);`

I'll fix it in the next commit

Reply:
Fixed